### ISSUE TITLE
Add string literal as __typename

### ISSUE THREAD

**Issue by felixfbecker** [avantcredit/gql2ts#84]
When doing a query like this
```graphql
query {
  search(text: "cat") {
    ... on Movie {
      director
    }
    ... on Book {
      author
    }
  }
}
```
on a schema like this
```graphql
union SearchResult = Movie | Book

type Query {
  search(text: String!): [SearchResult]!
}
```
you need to do a type check in the TS code to find out what type the results are, like this
```ts
if (item.__typename === 'Movie') {
  show(item.director)
} else {
  show(item.author)
}
```

TypeScript supports this through [discriminated unions](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions) if the schema in TS looks like this:
```ts
type SearchResult = Movie | Book
interface Movie {
  __typename: 'Movie'
  director: string
}
interface Book {
  __typename: 'Book'
  author: string
}
```
but it currently looks like this:
```ts
type SearchResult = Movie | Book
interface Movie {
  __typename: string
  director: string
}
interface Book {
  __typename: string
  author: string
}
```

**Comment by brettjurgens** [comment]
:+1: great idea. I can take a look tomorrow

**Comment by brettjurgens** [comment]
@felixfbecker just published 1.2.0 with this

**Comment by felixfbecker** [comment]
Awesome, thanks!
